sklearn/impute/tests/test_knn.py

```diff
@@ -513,8 +515,12 @@ def custom_callable(x, y, missing_values=np.nan, squared=False):
     assert_allclose(imputer.fit_transform(X), X_imputed)
 
 
+@pytest.mark.parametrize("working_memory", [None, 0])
 @pytest.mark.parametrize("na", [-1, np.nan])
-def test_knn_imputer_with_simple_example(na):
+# Note that we use working_memory=0 to ensure that chunking is tested, even
+# for a small dataset. However, it should raise a UserWarning that we ignore.
+@ignore_warnings(category=UserWarning)
```

Comment:
I would prefer `@pytest.mark.filterwarnings("ignore:partial warning message")
It allows to only match this specific warning (and make it a bit more specific at the same time)